### ISSUE TITLE
Using next-translate on the server

### ISSUE THREAD

**Issue by andrehsu** [vinissimus/next-translate#382]
Sometimes my API needs to send back messages using different locales. Is there way of doing that using next-translate? If not, is it on the roadmap?

**Comment by aralroca** [comment]
Next-translate translates both on client and server, but always within the (P)React environment. If you want to translate texts outside React, such as within an API route, right now there is no other way than moving the logic to React:

on API route:

```js
export default function handler(req, res) {
  const textToTranslate = { 
     key: 'api:example-of-message',
     params: { exampleOfParam: req.username }
  }

  res.statusCode = 200
  res.setHeader('Content-Type', 'application/json')
  res.end(JSON.stringify(textToTranslate))
}
```

on React:

```js
export default function useAPIMessage({ key, params  }) {
   const { t } = useTranslation() 
   return t(key, params)
}
```

It could be interesting to add the functionality in API route, but it would be good to understand the reasons why this is needed. Translations are part of the UI, why do you want to use translations within an API route instead of using the translations keys?

**Comment by andrehsu** [comment]
@aralroca The use case I had in mind was for my API to return the locale-translated error message for the user.

For example, Facebook has a `error_user_msg` in the API response for their Graph API.
https://developers.facebook.com/docs/graph-api/using-graph-api/error-handling/

It's uncommon, yes, but it's a use case. And I think the advantage of providing a `error_user_title` `error_user_msg` is that there is a single source of truth for user-visible error messages. If the API was being consumed somewhere else, then another set of error message translations would have to be maintained.

Also, in Next.js, if I want to translate some messages in getServerSideProps, which I currently can't do.

**Comment by aralroca** [comment]
Okay, it makes sense. Then I mark it as a future feature. Probably we can extend our API with something like:

```js
import serverSideT from 'next-translate/serverSideT'

// ...
export function getServerSideProps({ locale }) {
   const t = await serverSideT({ locale, namespace: "error" })
   const example = t('example-of-error-key')

   return { props: { example }  }
}
```

**Comment by aralroca** [comment]
@andrehsu I added `getT` 😊

# getT

**Size**: ~1.3kb 📦

Asynchronous function to load the `t` function outside components / pages. It works on both server-side and client-side.

Unlike the useTranslation hook, we can use here any namespace, it doesn't have to be a namespace defined in the "pages" configuration. It downloads the namespace indicated as a parameter on runtime.

Example inside `getStaticProps`:

```js
import getT from 'next-translate/getT'
// ...
export async function getStaticProps({ locale }) {
  const t = await getT(locale, 'common')
  const title = t('title')
  return { props: { title } }
}
```

Example inside API Route:

```js
import getT from 'next-translate/getT'

export default async function handler(req, res) {
  const t = await getT(req.query.__nextLocale, 'common')
  const title = t('title')

  res.statusCode = 200
  res.setHeader('Content-Type', 'application/json')
  res.end(JSON.stringify({ title }))
}
```

**Comment by aralroca** [comment]
@andrehsu I pre-released the change under **[1.0.0-canary.8](https://github.com/vinissimus/next-translate/releases/tag/1.0.0-canary.8)**. I would appreciate it if you could try it out on your project and give me feedback that everything is working well for you 😊 

If you wonder how to migrate to version 1.0, here is a guide:

- https://github.com/vinissimus/next-translate/blob/1.0.0-canary.8/docs/migration-guide-1.0.0.md

Any doubt, tell me. Thank you very much for reporting it 🙏 I guess that version 1.0 will be released soon.
